From e9309b036289d27a1bf469e7458bd5bfa2293161 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Wed, 5 Apr 2017 21:00:47 +0300
Subject: [PATCH 0950/1345] dts: a70x0: add device tree configuration for
 board setup D

commit  2023b6d0a06873969a48d8d620210e170374f4df from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

A7040 DB board is a modular board,
which supports multiple possible configurations.
This patch introduce predefined board topology for setup D.

- Add device tree for D setup
        - AP_UART, AP_SD
        - CP0_RGMII1, CP0_SDIO, CP0_I2C,
          CP0_SATA1, CP0_NAND

Change-Id: If713d4923b276db165a509a047c4551ba15d649f
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38544
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 .../mvebu/a7k-a8k/armada7040-db-setup.txt          |  134 ++++++++-------
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 arch/arm64/boot/dts/marvell/armada-7040-db-D.dts   |  180 ++++++++++++++++++++
 4 files changed, 256 insertions(+), 62 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-7040-db-D.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index e5a6a24..5378102 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -32,6 +32,9 @@ Boards:
    - "marvell,armada7040-db-E", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
+   - "marvell,armada7040-db-D", "marvell,armada7040-db"
+      when the board being used is the Armada 7040 Development board
+
    - "marvell,armada7040-db-nand", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
diff --git a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
index fb35e37..4c6374a 100644
--- a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
@@ -20,18 +20,18 @@ Board Setup
 ------------
 For A7040 A revision SoC - Please use armada-7040-db-A configuration,
 
-|		| 	1: A		|	2:B		|	3:C		|       4:E             |
-|---------------|-----------------------|-----------------------|-----------------------|-----------------------|
-|Device tree	|armada-7040-db-A.dtb	|armada-7040-db-B.dtb	|armada-7040-db-C.dtb	|armada-7040-db-E.dtb	|
-|CP0-SW1:1	| ON			| OFF			| ON			| OFF 			|
-|CP0-SW1:2	| OFF			| ON			| OFF			| OFF 			|
-|CP0-SW1:3	| ON			| ON			| ON			| ON			|
-|CP0-SW1:4	| OFF			| ON			| OFF			| OFF			|
-|CP0-SW1:5	| OFF			| ON			| OFF			| OFF			|
-|CP0-SW1:6	| ON			| ON			| ON			| ON			|
-|CP0-SW1:7	| ON			| ON			| ON			| ON			|
-|CP0-SW1:8	| OFF			| OFF			| OFF			| OFF			|
------------------------------------------------------------------------------------------------------------------
+|		| 	1: A		|	2:B		|	3:C		|       4:D             |       4:E             |
+|---------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
+|Device tree	|armada-7040-db-A.dtb	|armada-7040-db-B.dtb	|armada-7040-db-C.dtb	|armada-7040-db-D.dtb   |armada-7040-db-E.dtb	|
+|CP0-SW1:1	| ON			| OFF			| ON			| ON 			| OFF                   |
+|CP0-SW1:2	| OFF			| ON			| OFF			| OFF 			| OFF                   |
+|CP0-SW1:3	| ON			| ON			| ON			| ON			| ON                    |
+|CP0-SW1:4	| OFF			| ON			| OFF			| OFF			| OFF                   |
+|CP0-SW1:5	| OFF			| ON			| OFF			| OFF			| OFF                   |
+|CP0-SW1:6	| ON			| ON			| ON			| ON			| ON                    |
+|CP0-SW1:7	| ON			| ON			| ON			| ON			| ON                    |
+|CP0-SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF                   |
+----------------------------------------------------------------------------------------------------------------------------------------|
 
 DPR Setting
 ------------
@@ -43,36 +43,38 @@ DPR Setting
 | CP0 Serdess module (SLM1521)		| SW1 - (1,3,6,7)= ON,  (2,4,5,8)=OFF	 | SW1 - (2,3,4,5,6,7)= ON,  (1,8)=OFF    |
 | 					| DPR2,3,4,5 - 1-2 position		 | DPR2,3,4,5 - 2-3 position              |
 ----------------------------------------|----------------------------------------|----------------------------------------|
-
-|					| 	3: C				 |		4:E                                    |
-|---------------------------------------|----------------------------------------|-----------------------------------------------------|
-| Mpp Jumper Module (SLM1548)		| DPR133 - 143  2-3 position		 | DPR133 -143  1-2 position , DPR144-147 2-3 position |
-| Serdess Jumper Module (SLM1547)	| DPR152,153,154,155 - 1-2 position	 | DPR152,153,154,155 - 1-2 position                   |
-|					| DPR148,149,150,151 - 1-2 position	 | DPR148,149,150,151 - 1-2 position                   |
-| CP0 Serdess module (SLM1521)		| SW1 - (1,3,6,7)= ON,  (2,4,5,8)=OFF	 | SW1 - (3,6,7)= ON,  (1,2,4,5,8)=OFF                 |
-| 					| DPR2,3,4,5 - 1-2 position		 | DPR2,3,4,5 - 1-2 position                           |
-| Install TDM Module (SLM1448)		| on DB-7040 CON46			 | -						       |
-|					|					 |                                                     |
-| Install Audio Module (SLM1449)	| - 					 | -                                                   |
-|					|					 |                                                     |
-| DB-4070 Modular (SLM1546)		| - 					 | DPR53-54 1-2 position , DPR61 2-3 position          |
-|					| 					 | DPR10-11 position 1-2, DPR12-15 position 1-2        |
-----------------------------------------|----------------------------------------|-----------------------------------------------------|
+|					| 	3: C				 |	4:D					       |	5:E                                          |
+|---------------------------------------|----------------------------------------|-----------------------------------------------------|-----------------------------------------------------|
+| Mpp Jumper Module (SLM1548)		| DPR133 - 143  2-3 position		 | DPR138 - 143  2-3 position,DPR133-137 not connect   | DPR133 -143  1-2 position , DPR144-147 2-3 position |
+| Serdess Jumper Module (SLM1547)	| DPR152,153,154,155 - 1-2 position	 | DPR152,153,154,155 - 1-2 position                   | DPR152,153,154,155 - 1-2 position                   |
+|					| DPR148,149,150,151 - 1-2 position	 | DPR148,149,150,151 - 1-2 position		       | DPR148,149,150,151 - 1-2 position                   |
+| CP0 Serdess module (SLM1521)		| SW1 - (1,3,6,7)= ON,  (2,4,5,8)=OFF	 | SW1 - (1,3,6,7)= ON,  (1,2,4,5,8)=OFF               | SW1 - (3,6,7)= ON,  (1,2,4,5,8)=OFF                 |
+| 					| DPR2,3,4,5 - 1-2 position		 | DPR2,3,4,5 - 1-2 position                           | DPR2,3,4,5 - 1-2 position                           |
+|					|					 | CON9 not connect , CON10 connect,                   |                                                     |
+|					|					 | R77 R79 R81-82 R84-85 connect                       |                                                     |
+| Install TDM Module (SLM1448)		| on DB-7040 CON46			 | -                                                   | -                                                   |
+|					|					 |                                                     |                                                     |
+| Install Audio Module (SLM1449)	| - 					 | on DB-7040 CON46.                                   | -                                                   |
+|					|					 |                                                     |                                                     |
+|					|					 |                                                     |                                                     |
+| DB-4070 Modular (SLM1546)		| - 					 | wire from U36-1 to con47-2, DPR23 -NC               | DPR53-54 1-2 position , DPR61 2-3 position          |
+|					| 					 |                                                     | DPR10-11 position 1-2, DPR12-15 position 1-2        |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 
 The tables below summarizes the interface configuration of each setup
 
 SerDes Configuration
 ------------------------
 
-| CP0 Lane 	| 	1: A		|   2: B	| 3: C            | 4:E		  |
-|---------------|-----------------------|---------------|-----------------|---------------|
-| 0		| SGMII2		| PICE0 (x4)	| PCIE0 (x1)      |  PCIE0 (x1)   |
-| 1		| USB3_HOST0		| PCIE0 (x4)	| USB3_HOST0      |  SATA0        |
-| 2		| SFI (10G)		| PCIE0 (x4)	| SFI (10G)       |  SGMII0 (1G)  |
-| 3		| SATA1			| PCIE0 (x4)	| SATA1		  |  SATA1        |
-| 4		| USB3_HOST1		| USB3_HOST1	| USB3_HOST1      |  USB3_HOST1   |
-| 5		| PCIE2 (x1)		| PCIE2 (x1)	| PCIE2 (x1)      |  PCIE2 (x1)   |
--------------------------------------------------------------------------------------------
+| CP0 Lane 	| 	1: A		|   2: B	| 3: C            | 4:D		  | 5:E		  |
+|---------------|-----------------------|---------------|-----------------|---------------|---------------|
+| 0		| SGMII2		| PICE0 (x4)	| PCIE0 (x1)      |  PCIE0 (x1)   | PCIE0 (x1)	  |
+| 1		| USB3_HOST0		| PCIE0 (x4)	| USB3_HOST0      |  USB3_DEVICE0 | SATA0	  |
+| 2		| SFI (10G)		| PCIE0 (x4)	| SFI (10G)       |  SFI (1G)  	  | SGMII0 (1G)   |
+| 3		| SATA1			| PCIE0 (x4)	| SATA1		  |  SATA1        | SATA1	  |
+| 4		| USB3_HOST1		| USB3_HOST1	| USB3_HOST1      |  USB3_HOST1   | USB_HOST1	  |
+| 5		| PCIE2 (x1)		| PCIE2 (x1)	| PCIE2 (x1)      |  PCIE2 (x1)   | PCIE2 (x1)	  |
+-----------------------------------------------------------------------------------------------------------
 
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
@@ -81,31 +83,33 @@ SerDes Configuration
 Multi-purpose pin configurations
 --------------------------------
 
-Setup  #	|	1	|	2	|       3       |	4	|
-Name   #	|  	A 	|	B	|       C       |	E	|
----------------------------------------------------------------------------------
-AP806:				|		|		|		|
----------------------------------------------------------------------------------
-	AP-SDIO | [0-5]		| [0-5]		| [0-5]         | [0-3]		|
-	AP-SPI0 | N/C		| N/C		| N/C		| [4,5]		|
-	AP-UART0| [11,19]	| [11,19]	| [11,19]       | [11,19]	|
----------------------------------------------------------------------------------
-CP:				|		|		|		|
----------------------------------------------------------------------------------
-	CP-SPI0	| N/C		| N/C		| [6-11]	| N/C		|
-	CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| [13-16]	|
-	NAND	| N/C		| N/C		| N/C		| N/C		|
-	RGMII0	| N/C		| [44-55]	| [0-11]	| [0-11]	|
-	RGMII1	| [0-11]	| [0-11]	| [44-55]	| N/C		|
-	CP-UART0| [29-30]	| [29-30]	| [29-30	| [29-30|	|
-	CP-UART1| N/C		| N/C		| N/C		| [47,49-51]	|
-	SD	| [56-61]	| [56-61]	| [56-61]	| [54,56-61]	|
-	GPIO	| [62]		| [62]		| [62]		| [62]		|
-	TDM	| N/C		| N/C		| [0-5]		| N/C		|
-	TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
-	SATA0	| N/C		| N/C		| N/C		| N/C		|
-	SATA1	| [28]		| [28]		| [28]		| [28]		|
----------------------------------------------------------------------------------
+Setup  #	|	1	|	2	|       3       |	4	|	5	|
+Name   #	|  	A 	|	B	|       C       |	D	|	E	|
+-------------------------------------------------------------------------------------------------
+AP806:				|		|		|		|		|
+-------------------------------------------------------------------------------------------------
+	AP-SDIO | [0-5]		| [0-5]		| [0-5]         | [0-5]		| N/C		|
+	AP-SPI0 | N/C		| N/C		| N/C		| N/C		| [0-3]		|
+	AP-I2C	| N/C		| N/C		| N/C		| N/C		| [4,5]		|
+	AP-UART0| [11,19]	| [11,19]	| [11,19]       | [11,19]	| [11,19]	|
+-------------------------------------------------------------------------------------------------
+CP:				|		|		|		|		|
+-------------------------------------------------------------------------------------------------
+	CP-SPI0	| N/C		| N/C		| [6-11]	| N/C		| N/C		|
+	CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	|
+	NAND	| N/C		| N/C		| N/C		| [15-27]	| N/C		|
+	RGMII0	| N/C		| [44-55]	| [0-11]	| N/C		| [0-11]	|
+	RGMII1	| [0-11]	| [0-11]	| [44-55]	| [44-55]	| N/C		|
+	CP-UART0| [29-30]	| [29-30]	| [29-30	| [29-30]	| [29-30|	|
+	CP-UART1| N/C		| N/C		| N/C		| N/C		| [47,49-51]	|
+	SD	| [56-61]	| [56-61]	| [56-61]	| [56-61]	| [54,56-61]	|
+	GPIO	| [62]		| [62]		| [62]		| [62]		| [62]		|
+	TDM	| N/C		| N/C		| [0-5]		| N/C		| N/C		|
+	TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
+	SATA0	| N/C		| N/C		| N/C		| N/C		| N/C		|
+	SATA1	| [28]		| [28]		| [28]		| [28]		| [28]		|
+	USB_DEV | N/C 		| N/C 		| N/C		| [7]		| N/C		|
+-------------------------------------------------------------------------------------------------
 
 Network configuration
 ---------------------
@@ -130,8 +134,14 @@ Network configuration
  eth0           | CP0   | eth0          | 0             | SFI (10G)                     |
  eth1           | CP0   | eth1          | 3             | RGMII1                        |
  ----------------------------------------------------------------------------------------
+4: D:
+ Interface      | CP#   | PPv2 Port     | GMAC Port     | Board Interface               |
+ ---------------|-------|---------------|---------------|-------------------------------|
+ eth0           | CP0   | eth0          | 0             | SFI (10G)                     |
+ eth1           | CP0   | eth1          | 3             | RGMII1                        |
+ ----------------------------------------------------------------------------------------
 
-4: E:
+5: E:
  Interface      | CP#   | PPv2 Port     | GMAC Port     | Board Interface               |
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0           | CP0   | eth0          | 0             | SGMII0 (1G)                   |
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 8fad107..06d3fa9 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -13,6 +13,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-C.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-E.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
new file mode 100644
index 0000000..7a58664
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 development board
+ * This board file supports the D configuration of the board
+ */
+
+#include "armada-7040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-7040 development board D setup";
+	compatible = "marvell,armada7040-db-D", "marvell,armada7040-db", "marvell,armada7040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+				clock-frequency = <100000>;
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			sata: sata@540000 {
+				status = "okay";
+
+				sata-port@0 {
+					status = "disabled";
+				};
+				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			usb3h0: usb3@500000 {
+				status = "disabled";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			u3d@520000 {
+				status = "okay";
+			};
+			udc@524100 {
+				status = "okay";
+			};
+			nand@720000 {
+				status = "okay";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			spi@700680 {
+				status = "disabled";
+			};
+			i2c1: i2c@701000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+			i2c2: i2c@701100 {
+				status = "disabled";
+			};
+
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /*lane-2*/
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "none";
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x620000 {
+			status = "disabled";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
-- 
1.7.9.5

